components/steering_wheel/HW/include/HW_gpio.h

```diff
@@ -16,12 +16,32 @@ extern "C" {
 #include "SystemConfig.h"
 
 
+/******************************************************************************
+ *                             T Y P E D E F S
+ ******************************************************************************/
+
+typedef enum
+{
+    HW_PIN_RESET = GPIO_PIN_RESET,
+    HW_PIN_SET   = GPIO_PIN_SET,
+} HW_GPIO_PinState_E;
+
+typedef GPIO_TypeDef HW_GPIO_TypeDef;
+
+typedef struct
+{
+    HW_GPIO_TypeDef* port;
+    uint16_t         pin;
+} HW_GPIO_Input_S;
+
+
 /******************************************************************************
  *            P U B L I C  F U N C T I O N  P R O T O T Y P E S
  ******************************************************************************/
 
 void MX_GPIO_Init(void);
 
+HW_GPIO_PinState_E HW_GPIO_ReadPin(const HW_GPIO_Input_S *input);

```

Comment:
nit: remove `HW_` prefix, just GPIO is fine I think

Reply:
I added HW_ to explicitly show that it is tied to the firmware interface. I think its beneficial long term just as HAL in regular stm32 programming shows its interfacing with hardware. Thoughts?

Reply:
I think GPIO already indicates that pretty clearly